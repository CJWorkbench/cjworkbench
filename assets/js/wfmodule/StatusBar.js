// ---- StatusBar ----

import React from "react";
import PropTypes from "prop-types";

export default class StatusBar extends React.Component {
  render() {

    let barColor = undefined;

    switch (this.props.status) {
      case 'ready':
        barColor = (this.props.isSelected) ? 'module-output-bar-blue' : 'module-output-bar-white'
        break;
      case 'busy':
        barColor = 'module-output-bar-orange';
        break;
      case 'error':
        barColor = (this.props.isSelected) ? 'module-output-bar-red' : 'module-output-bar-pink'
        break;
      default:
        barColor = 'module-output-bar-white';
        break;
    }

    return <div className={barColor} />
  }
}

StatusBar.PropTypes = {
  isSelected: PropTypes.bool,
  status: PropTypes.string
};