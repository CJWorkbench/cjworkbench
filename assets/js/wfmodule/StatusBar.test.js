import React from 'react'
import StatusBar from './StatusBar'
import { shallow } from 'enzyme'

describe('Status bar', () => {
  it('Renders the error color', () => {
    let wrapper = shallow(
      <StatusBar
        isSelected={true}
        status="error"
      />
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('div').first().props().className).toEqual("module-output-bar-red");
  });

  it('Renders the unselected error color', () => {
    let wrapper = shallow(
      <StatusBar
        isSelected={false}
        status="error"
      />
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('div').first().props().className).toEqual("module-output-bar-pink");
  });

  it('Renders the busy color', () => {
    let wrapper = shallow(
      <StatusBar
        isSelected={false}
        status="busy"
      />
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('div').first().props().className).toEqual("module-output-bar-orange");
  });

  it('Renders the selected ready color', () => {
    let wrapper = shallow(
      <StatusBar
        isSelected={true}
        status="ready"
      />
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('div').first().props().className).toEqual("module-output-bar-blue");
  });

  it('Renders the unselected ready color', () => {
    let wrapper = shallow(
      <StatusBar
        isSelected={false}
        status="ready"
      />
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('div').first().props().className).toEqual("module-output-bar-white");
  });

  it('Renders the default color', () => {
    let wrapper = shallow(
      <StatusBar
        isSelected={false}
        status="garbage_return_status"
      />
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('div').first().props().className).toEqual("module-output-bar-white");
  });
});