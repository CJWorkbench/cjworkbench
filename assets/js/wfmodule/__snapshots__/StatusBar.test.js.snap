// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Status bar Renders the busy color 1`] = `
<div
  className="module-output-bar-orange"
/>
`;

exports[`Status bar Renders the default color 1`] = `
<div
  className="module-output-bar-white"
/>
`;

exports[`Status bar Renders the error color 1`] = `
<div
  className="module-output-bar-red"
/>
`;

exports[`Status bar Renders the selected ready color 1`] = `
<div
  className="module-output--live"
/>
`;

exports[`Status bar Renders the unselected error color 1`] = `
<div
  className="module-output-bar-pink"
/>
`;

exports[`Status bar Renders the unselected ready color 1`] = `
<div
  className="module-output-bar-white"
/>
`;
